cmd/system/caddy.go

```diff
@@ -0,0 +1,157 @@
+		}
+
+		arch, osVer := env.GetClientArch()
+
+		// TODO: Check if it's ubuntu or debian
+		if strings.ToLower(osVer) != "linux" {
+			return fmt.Errorf("this app only supports Linux")
+		}
+
+		if cmd.Flags().Changed("arch") {
+			arch, _ = cmd.Flags().GetString("arch")
+		}
+
+		dlArch := arch
+		if arch == "x86_64" {
+			dlArch = "amd64"
+		} else if arch == "aarch64" {
+			dlArch = "arm64"
+		} else {
+			dlArch = arch
+		}
+
+		if version == "" {
+			v, err := get.FindGitHubRelease("caddyserver", "caddy")
+			if err != nil {
+				return err
+			}
+			version = v
+		} else if !strings.HasPrefix(version, "v") {
+			version = "v" + version
+		}
+
+		fmt.Printf("Installing version: %s for: %s\n", version, dlArch)
+
+		filename := fmt.Sprintf("caddy_%s_linux_%s.tar.gz", strings.TrimPrefix(version, "v"), dlArch)
+		dlURL := fmt.Sprintf(githubDownloadTemplate, "caddyserver", "caddy", version, filename)
+
+		fmt.Printf("Downloading from: %s\n", dlURL)
+
+		progress, _ := cmd.Flags().GetBool("progress")
+		outPath, err := get.DownloadFileP(dlURL, progress)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(outPath)
+
+		fmt.Printf("Downloaded to: %s\n", outPath)
+
+		f, err := os.OpenFile(outPath, os.O_RDONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+
+		fmt.Printf("Unpacking Caddy to: %s\n", path.Join(installPath, "caddy"))
+
+		if err := archive.Untar(f, installPath, true, true); err != nil {
+			return err
+		}
+
+		svcTmpPath, err := get.DownloadFileP("https://raw.githubusercontent.com/caddyserver/dist/master/init/caddy.service", false)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Downloaded caddy.service file to %s\n", svcTmpPath)
+		defer os.Remove(svcTmpPath)
+
+		caddySystemFile := "/etc/systemd/system/caddy.service"
+		_, err = get.CopyFile(svcTmpPath, caddySystemFile)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Copied caddy.service file to %s\n", caddySystemFile)
+
+		caddyHomeDir := "/var/lib/caddy"
+		os.MkdirAll(caddyHomeDir, fs.ModeDir)
+		caddyConfDir := "/etc/caddy"
+		os.MkdirAll(caddyConfDir, fs.ModeDir)
+		fmt.Printf("Created caddy home %s and Conf %s directory\n", caddyHomeDir, caddyConfDir)
+		// TODO: Add empty CaddyFile and if required FAAS_DOMAIN setup too
+
+		shellCmds := []string{
+			"$SUDO chown --recursive caddy:caddy /var/lib/caddy",
+			"$SUDO chown --recursive caddy:caddy /etc/caddy",
+			"$SUDO systemctl enable caddy",
+			"$SUDO systemctl daemon-reload",
+			"$SUDO systemctl start caddy",
+		}
+
+		caddyUser := "caddy"
+		if _, err = user.Lookup(caddyUser); errors.Is(err, user.UnknownUserError(caddyUser)) {
+			addUserCmd := fmt.Sprintf("$SUDO useradd --system --home %s --shell /bin/false %s", caddyHomeDir, caddyUser)
+			shellCmds = append([]string{addUserCmd}, shellCmds...)
+
+			fmt.Printf("User created for caddy server.\n")
+		}
+
+		commands := strings.Join(shellCmds, "; ")
+		if err = exec.Command(commands).Run(); err != nil {
```

Comment:
Could you try go-execute which is used elsewhere? It has a shell option if needed.

Reply:
Done